Shared/Samples/Snap geometry edits/SnapGeometryEditsView.GeometryEditorModel.swift

```diff
@@ -0,0 +1,155 @@
+    /// An object that acts as a view model for the geometry editor menu.
+    @MainActor
+    class GeometryEditorModel: ObservableObject {
+        /// The geometry editor.
+        let geometryEditor = GeometryEditor()
+        
+        /// The graphics overlay used to save geometries to.
+        let graphicsOverlay = GraphicsOverlay(renderingMode: .dynamic)
+        
+        /// A Boolean value indicating if the geometry editor can perform an undo.
+        @Published private(set) var canUndo = false
+        
+        /// A Boolean value indicating if the geometry editor can perform a redo.
+        @Published private(set) var canRedo = false
+        
+        /// The currently selected element.
+        @Published private(set) var selection: GeometryEditorElement?
+        
+        /// A Boolean value indicating if the geometry can be saved to a graphics overlay.
+        @Published private(set) var canSave = false
+        
+        /// A Boolean value indicating if the geometry can be cleared from the geometry editor.
+        @Published private(set) var canClearCurrentSketch = false
+        
+        /// A Boolean value indicating if the saved sketches can be cleared.
+        @Published private(set) var canClearSavedSketches = false
+        
+        /// The current geometry of the geometry editor.
+        @Published private(set) var geometry: Geometry? {
+            didSet {
+                canUndo = geometryEditor.canUndo
+                canRedo = geometryEditor.canRedo
+                canClearCurrentSketch = geometry.map { !$0.isEmpty } ?? false
+                canSave = geometry?.sketchIsValid ?? false
+            }
+        }
+        
+        /// A Boolean value indicating if the geometry editor has started.
+        @Published private(set) var isStarted = false
+        
+        /// A Boolean value indicating if the scale mode is uniform.
+        @Published var shouldUniformScale = false {
+            didSet {
+                configureGeometryEditorTool(geometryEditor.tool, scaleMode: scaleMode)
+            }
+        }
+        
+        /// The scale mode to be set on the geometry editor.
+        private var scaleMode: GeometryEditorScaleMode {
+            shouldUniformScale ? .uniform : .stretch
+        }
+        
+        /// Creates the geometry menu with a geometry editor.
+        init() {
+            Task { [weak self, geometryEditor] in
+                for await geometry in geometryEditor.$geometry {
+                    self?.geometry = geometry
+                }
+            }
+            Task { [weak self, geometryEditor] in
+                for await selection in geometryEditor.$selectedElement {
+                    self?.selection = selection
+                }
+            }
+        }
+        
+        /// Saves the current geometry to the graphics overlay and stops editing.
+        /// - Precondition: `canSave`
+        func save() {
+            precondition(canSave)
```

Comment:
Was added here dbabd11fdfa3b191588a84f296bcd914ea66878c see [this](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/129#discussion_r1146548483) comment.

I think it was added to force crash when a programmer error happens, i.e., save is called when the geometry is not valid. I thought just force unwrap the geometry should also be OK, but it may provide less crash info than the precondition?